integration/nighthawk_test_server.py

```diff
@@ -0,0 +1,90 @@
+import json
+import logging
+import os
+import socket
+import subprocess
+import sys
+import tempfile
+import threading
+import time
+from string import Template
+
+from common import IpVersion
+
+
+class TestServerBase(object):
+  """
+    Base class for running a server in a separate process.
+    """
+
+  def __init__(self, server_binary_path, config_template_path, server_ip, server_port, ip_version,
+               server_binary_config_path_arg, parameters):
+    self.ip_version = ip_version
+    self.server_binary_path = server_binary_path
+    self.config_template_path = config_template_path
+    self.server_thread = threading.Thread(target=self.serverThreadRunner)
+    self.server_process = None
+    self.server_ip = server_ip
+    self.socket_type = socket.AF_INET6 if ip_version == IpVersion.IPV6 else socket.AF_INET
+    self.server_port = server_port
+    self.parameters = parameters
+    self.server_binary_config_path_arg = server_binary_config_path_arg
+
+  def serverThreadRunner(self):
+    self.parameters["server_ip"] = self.server_ip
+    self.parameters["server_port"] = self.server_port
+
+    with open(self.config_template_path) as f:
+      config = Template(f.read())
+      config = config.substitute(self.parameters)
+
+    parameterized_config_path = None
+    with tempfile.NamedTemporaryFile(mode="w", delete=False, suffix=".yaml") as tmp:
+      parameterized_config_path = tmp.name
+      tmp.write(config)
```

Comment:
Arguably we could use the bootstrap CLI here to avoid needing to write files; I don't think it makes a huge difference, but throwing it out there.

Reply:
Actually in earlier revisions of this it worked via the bootstrap CLI; but like with the port, I leaned toward doing it like this with supporting for servers in the future in mind.